db/migrate/20160317142959_add_index_to_switch.rb

```diff
@@ -0,0 +1,5 @@
+class AddIndexToSwitch < ActiveRecord::Migration[5.0]
+  def change
+    add_index :switches, :uid_ems, :unique => true
```

Comment:
I don't think host_id|switch_name is going to be unique in this table.  You can easily have 'host-2|vswitch0' from two VCs.  I'm not sure that we care about being able to identify a switch across VCs though so might not need the unique constraint...